### PR TITLE
Add group ACL rule for membership in another group

### PR DESCRIPTION
**Pull request by jboning**
This will make it possible to configure different API key requirements for different alliances.

### CODE REVIEW

File: brave/core/group/controller.py [hunk 1]
```diff
@@ -129,20 +129,33 @@ def listify(rule, field):
...
-        log.debug("really!")
         if rule_set == "request":
             self.group.request_rules = rule_objects
         elif rule_set == "join":
             self.group.join_rules = rule_objects
         else:
             self.group.rules = rule_objects
+
+        try:
+            self.group.recursion_check()
```
Comment (Acidity) [inline brave/core/group/controller.py]: I believe this should be self.group.cycle_check()

**Comment by Acidity** [comment]
When attempting to make a cyclic reference, the ACL check from the server just displays [object Object] rather than an error code.

**Comment by Acidity** [comment]
Another thought, since this only lets you specify one group in the ACL, if you want to do a DENY OR or a GRANT AND on multiple groups you have to make a new group just for that, which isn't great. Not sure how easy it would be to make the ACL support multiple groups though.

**Comment by Acidity** [comment]
Also: deleting a group that another group refers to without deleting the ACL rule first breaks pretty much everything.

**Comment by jboning** [comment]
Note to self, don't merge yet, this can still result in dangling references if you change the ID of a referenced group.

**Pull request merged**